Hazel.UnitTests/SocketCapture.cs

```diff
@@ -126,44 +136,50 @@ private void ReceiveLoop()
 
         private void SendToRemoteLoop()
         {
-            foreach (ByteSpan packet in this.forRemote.GetConsumingEnumerable())
+            while (!this.cancellationToken.IsCancellationRequested)
             {
-                if (this.cancellationToken.IsCancellationRequested)
-                {
-                    break;
-                }
-
                 if (this.SendToRemoteSemaphore != null)
                 {
-                    if (!this.SendToRemoteSemaphore.WaitOne(100))
+                    if (!this.SendToRemoteSemaphore.WaitOne(this.DelayBeforeDiscardingMs))
                     {
                         continue;
                     }
                 }
 
-                this.captureSocket.SendTo(packet.GetUnderlyingArray(), packet.Offset, packet.Length, SocketFlags.None, this.remoteEndPoint);
+                if (this.forRemote.TryTake(out var packet))
```

Comment:
Oops, this breaks tests, we don't discard packets when waiting too long.

Reply:
hmmm do we need to loop through the collection and discard any that time out? if all time-outs are the same length, I guess you could check the longest waiting packet only as well rather than looping through

Reply:
I decided to make packet dropping not based on time. This 100 drop time was kind of a hidden feature that I didn't even know I was breaking at first. The comments elsewhere say that 1000ms causes packet resends and that makes sense, but intuitively you might assume the original packets delayed, not dropped.

If we took this a step further, I'd have a SocketCapture.WaitForPacketToLocalCount(int) and there'd be no need for Thread.Sleep at all. But the lock sequencing gets kinda tricky, so for now we are just assuming that for each Handshake step, we will wait for the preceding step. Any time we flush the packet queue, it will be just one kind of flight of packets. If we don't wait long, should be just one flight. But if we wait long, we must flush the whole queue.

Reply:
Cool thanks, I understand the motivation to make it not time based in a hidden way. Makes sense to want to control drop vs delay

I can see how these more "networked integration tests" are more complicated as we need to induce the resend rather than a more close-knit test that might just mock send a certain packet sequence one at a time, no waiting in-between

Yeah, I'm a little conflicted here in opinion as I feel like what we want is tight control over each packet, which I think is the further steps you are thinking, but since resends are kind of not controlled, we have to make guesses..

Reply:
Yeah, I think I have improved it in that it'll be easier going forward, but really it is just no worse than it was before. So for this PR. I'll probably call it.

Reply:
Yeah, agree with that - I think I see more specifically where I'm thinking and will comment on that line, but overall yeah, this is likely a good spot for this PR